services/restaurants/restaurant_comparer.py

```diff
@@ -0,0 +1,143 @@
+from typing import Dict
+from models.restaurants_model import Restaurant
+from constants import API_TOKEN
+from yelp_fusion_api.yelp_fusion import YelpFusion
+import nltk
+from nltk.corpus import wordnet
+import re
+
+class RestaurantComparer:
+
+    def __init__(self):
+        self.curr_resturant = None
+
+    def compare_restaurants(self, rest1, rest2):
+        # takes in restaurant 1 id & restaurant 2 id
+        headers = {
+            "Authorization": f"Bearer {API_TOKEN}"
+        }
+        # finds restaurant info
+        restInfo1 = YelpFusion().find_restaurant_by_id(id=rest1, headers=headers)
+        restInfo2 = YelpFusion().find_restaurant_by_id(id=rest2, headers=headers)
+
+        similarity_score = 0
+        max_score = 0
+
+        # compares price, number of reviews, rating, and cuisine type
```

Comment:
These 4 similarity scorer code blocks can be made into 1 dynamic function since the math being performed is the same for each score. Reduce this code and make it into a dynamic function.